packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/workflow/Workflow.java

```diff
@@ -19,21 +29,51 @@
 @EqualsAndHashCode(callSuper = true)
 @Data
 @Accessors(chain = true)
-@TSModel
-public class Workflow extends TerariumAsset implements Serializable {
+@Entity
+public class Workflow extends TerariumAsset {
 
 	@Serial
 	private static final long serialVersionUID = -1565930053830366145L;
 
-
+	@Schema(defaultValue = "My New Workflow")
 	private String name;
 
 	private String description;
 
 	private Transform transform;
 
-	private Object nodes;
+	@Convert(converter = JsonConverter.class)
+	@JdbcTypeCode(SqlTypes.JSON)
+	private List<WorkflowNode> nodes;
+
+	@Convert(converter = JsonConverter.class)
+	@JdbcTypeCode(SqlTypes.JSON)
+	private List<WorkflowEdge> edges;
+
+	public Workflow clone() {
+		final Workflow clone = new Workflow();
+		clone.setId(UUID.randomUUID());
```

Comment:
If this hasn't been saved in postgres, do we want to assign it a UUID at all? I thought that the UUID was auto generated at creation or insertion

Reply:
Oooo shoot you are right, this is gonna be a bit tedious: Workflows have nested fields at refer to the parents ID. So we would need to write it out, get the created value and re-write it out after populating the nested fields. Lemme see if there's a more elegant way to get around this.